dss/operations/checkout.py

```diff
@@ -63,72 +62,72 @@ def _sleepy_checkout(self, function=None, **kwargs):
 
-    def _parse_key(self, key):
+    @staticmethod
+    def _parse_key(key):
         try:
             version = VERSION_REGEX.search(key).group(0)
             uuid = UUID_REGEX.search(key).group(0)
         except IndexError:
-            raise RuntimeError(f'unable to parse the key {key}')
+            raise RuntimeError(f'Unable to parse the key: {key}')
         return uuid, version
 
     def process_keys(self):
         raise NotImplementedError()
 
     def __call__(self, argv: typing.List[str], args: argparse.Namespace):
-        if self.keys is not None:
-            self.process_keys()
+        self.process_keys()
 
 
 checkout = dispatch.target("checkout", help=__doc__)
 
 
 @checkout.action("remove",
                  arguments={"--replica": dict(choices=[r.name for r in Replica], required=True),
-                            "--keys": dict(nargs="+", help="keys to remove from checkout", required=True)})
-class remove(CheckoutHandler):
+                            "--keys": dict(nargs="+", help="Keys to remove from checkout.", required=True)})
+class Remove(CheckoutHandler):
     def process_keys(self):
-        """
-        Remove a bundle from the checkout bucket
-        """
+        """Remove keys from the checkout bucket."""
         for _key in self.keys:
             if DSS_BUNDLE_KEY_REGEX.match(_key):
-                for key in self.handle.list(self.bucket, _key):  # handles checkout/bundle/*
-                    self._verify_delete(self.handle, self.bucket, key)
+                for key in self.handle.list(self.checkout_bucket, _key):  # handles checkout/bundle/*
+                    self._verify_delete(self.handle, self.checkout_bucket, key)
                 uuid, version = self._parse_key(_key)
                 manifest = get_bundle_manifest(replica=self.replica, uuid=uuid, version=version)
                 if manifest is None:
-                    logger.warning(f"unable to locate manifest for fqid: {self.bucket}/{_key}")
+                    logger.warning(f"Unable to locate manifest for fqid: {self.checkout_bucket}/{_key}")
                     continue
                 for _files in manifest['files']:
                     key = compose_blob_key(_files)
-                    self._verify_delete(self.handle, self.bucket, key)
+                    self._verify_delete(self.handle, self.checkout_bucket, key)
             elif _key.startswith(FILE_PREFIX):
                 # should handle other keys, files/blobs
-                file_metadata = self._verify_get(self.handle, self.replica.bucket, _key)
-                self._verify_delete(self.handle, self.bucket, key=compose_blob_key(file_metadata))
+                file_metadata = self._get_metadata(self.handle, self.replica.bucket, _key)
+                self._verify_delete(self.handle, self.checkout_bucket, key=compose_blob_key(file_metadata))
+            else:
+                print(f'INVALID KEY REGEX: {_key}')
```

Comment:
i would avoid printing stuff that does not need to be printed, unless you can mute it, If i want to pipe this into JQ, the print statement causes problems.

Reply:
Hmmm... should we raise here and disallow nonsense keys?  I can see your point when wanting to pipe.  Inputting a key like `borkel/snorkel`, I expect to either see that it doesn't exist in the checkout bucket or that it raises an error because it does not match the expected key.

Reply:
perhaps https://github.com/HumanCellAtlas/data-store/pull/2549#discussion_r340356914

Reply:
Warnings written to [sys.stderr](https://docs.python.org/3/library/sys.html#sys.stderr) will not contaminate stdout.

Reply:
@xbrianh I like that idea, will implement.